File: cfrac/gcd.pl
#!/usr/bin/perl

print "gcd(numerator,denominator)\ngcd(";
$gcd = <STDIN>;
chomp($gcd); chomp($gcd);
@int = split(/,/,$gcd);

chomp($int[1]);
$num=$int[1]; $den=$int[0]; $rem=1;
while($rem!=0){
	$prev=$rem;
	$rem=$num % $den;
	$num=$den;
	$den=$rem;
}
print "The Greatest Common Divisor gcd($int[0],$int[1]=$prev\n";
print "The Simplest form of $int[0] / $int[1] is\n";
print $int[0]/$prev," / ",$int[1]/$prev;
<STDIN>;

end;

File: cfrac/gcd.c
#include<stdio.h>
#include<stdlib.h>

void main(){
	int n,d,num,den,rem,prev;
	printf("gcd(N,D)\nN=");
	scanf("%d",&n);
	printf("D=");
	scanf("%d",&d);
	
	num=n; den=d; rem=1;
	while(rem!=0){
		prev=rem;
		rem=num % den;
		num=den;
		den=rem;
	}
	printf("The Greatest Common Divisor gcd(%d,%d)=%d\n",n,d,prev);
	printf("The Simplest form of %d / %d is \n",n,d);
	printf("%d / %d\n",n/prev,d/prev);
	
	system("pause");
}
